### DISCUSSION TITLE
How OMNeT++ implements runtime configuration

### DISCUSSION THREAD

**Discussion by zhangteng0526** [omnetpp/omnetpp#1013] [Q&A] [answered]
For example, my simulated network is an AODV network. Usually the communication between nodes is pre-set in the ini file or ScenarioManager. Now suppose I suddenly have an idea during the simulation (which cannot be done in the above method) to be able to set node 1 to send a message to node 2 at some point while the simulation is running. Please let me know if this idea is possible and how. Hope to get your help, thank you here! ! !

**Comment by rhornig** [comment]
You write a simple module in C++. You can implement any logic that suits your needs. All the various applications and the scenario manager is just a C++ class in the end. They expose some parameters or configuration files to fine tune their behavior, but you can do your own if that does not fit your needs.

**Reply by zhangteng0526** [reply]
But I can't figure out what should I do to achieve such functionality (I haven't decided in advance, at some point during the simulation I suddenly want to send a message from one node to another)。Do you have any ideas and suggestions, thank you

**Reply by rhornig** [reply, marked as answer]
You mean you want to do a message send based on USER interaction (i.e. interactively)? 

This is called real-time/hardware in the loop simulation where some events from **outside** of the simulation triggers some events inside the simulation. Check the sockets sample in OMNeT++. It does exactly like that. Listens on an opsystem socket and send data received on that socket into the simulation (and vice versa).

How you communicate with the simulation is up to you. You may chose sockets, pipes, or you can even poll for the presence of a file and get the external event trigger from there.
